### PR TITLE
Fixes Issue#106

### PR DESCRIPTION
**Pull request by ornemarioni**
This is an alternative to solve issue #106

### CODE REVIEW

**Comment by ornemarioni** [comment]
I need some help with potential validation, don't pass the test @leliel12 @BrunoSanchez @vcristiani @nvillagra @RafaelPignata622 @antotaverna

**Comment by antotaverna** [comment]
In test_GCKmeans, I change the assert equal for an assert_allclose and works. I don't know why not all elements are exactly equals

np.testing.assert_allclose(result, expected, rtol=7.0e-9, atol=1.7e-07)

File: galaxychop/core.py [hunk 1]
```diff
@@ -189,6 +208,34 @@ def __attrs_post_init__(self):
...
 
+    def __potential_energy_init__(self):
+        """
+        Potential energy input validator.
+
+        This method determine the validation of input of the specific
+        potential energy.
+        """
+        if np.any(self.arr_.pot_dm != 0.0) and (
+            np.all(self.arr_.pot_s == 0.0) or np.all(self.arr_.pot_s == 0.0)
```
Comment (vcristiani) [inline galaxychop/core.py:219]: np.all(self.arr_.pot_s == 0.0) or np.all(self.arr_.pot_**g** == 0.0)

File: galaxychop/core.py [hunk 2]
```diff
@@ -189,6 +208,34 @@ def __attrs_post_init__(self):
...
+        ):
+            raise ValueError(
+                "Potential energy must be instanced for all type particles"
+            )
+
+        if np.any(self.arr_.pot_s != 0.0) and (
+            np.all(self.arr_.pot_dm == 0.0) or np.all(self.arr_.pot_s == 0.0)
```
Comment (vcristiani) [inline galaxychop/core.py:226]: np.all(self.arr_.pot_dm == 0.0) or np.all(self.arr_.pot_**g** == 0.0)

**Comment by vcristiani** [comment]
All looks good, merge.

**Review by BrunoSanchez — APPROVED** [review]
Approved. Reviewed code structure and new implemented tests.
Have not checked or validated physically the models.

**Pull request merged**